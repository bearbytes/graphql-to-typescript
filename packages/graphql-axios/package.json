{
  "publishConfig": {
    "access": "public"
  },
  "name": "@bearbytes/graphql-axios",
  "description": "",
  "version": "0.0.17-alpha.0",
  "author": "",
  "license": "ISC",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc",
    "prepare": "npm run build"
  },
  "dependencies": {
    "@bearbytes/graphql-to-typescript": "^0.0.17-alpha.0",
    "axios": "^0.18.0"
  },
  "devDependencies": {
    "@types/graphql": "^14.2.0"
  },
  "gitHead": "6aa72d9f6673cd06cb3d42625d5c2f3e8a839f05"
}
