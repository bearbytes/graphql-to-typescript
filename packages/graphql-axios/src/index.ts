import { Client, OperationOptions, OperationResult, Config } from './types'
import axios, { AxiosRequestConfig } from 'axios'
import { Query, Mutation } from '@bearbytes/graphql-to-typescript'

export * from './types'

export function createClient<T>(
  typedGraphQL: T,
  clientConfig: Config
): Client<T> {
  return {
    query: executeOperation,
    mutate: executeOperation,
    subscribe: () => {
      throw Error('not yet implemented')
    },
  }

  async function executeOperation<Name extends Query<T> | Mutation<T>>(
    operationName: Name,
    options: OperationOptions<T, Name>
  ): Promise<OperationResult<T, Name>> {
    const { url } = clientConfig

    const body = {
      operationName,
      query: typedGraphQL[operationName],
      variables: options.variables,
    }

    const response = await axios.post(
      url,
      body,
      {
        headers: {
          'Content-Type': 'application/json',
        },
      }
      // axiosConfig
    )
    return response
  }
}
