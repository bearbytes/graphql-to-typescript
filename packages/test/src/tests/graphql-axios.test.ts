import typedGraphQL from '../outputs/output'
import { createClient } from '@bearbytes/graphql-axios'
import {
  Query,
  OperationData,
  Mutation,
  Subscription,
  OperationVariables,
} from '@bearbytes/graphql-to-typescript'
import { startServer } from '../graphql-server'

const client = createClient<typeof typedGraphQL>(typedGraphQL, {
  url: 'http://localhost:5000',
})

test('simple query', async () => {
  const response = await client.query('simpleQuery', { variables: {} })
  expect(response).toMatchSnapshot()
})
