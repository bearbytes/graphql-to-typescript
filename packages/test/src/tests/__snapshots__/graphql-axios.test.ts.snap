// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple query 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "data": "{\\"operationName\\":\\"simpleQuery\\",\\"query\\":\\"query simpleQuery {\\\\n  scalarString\\\\n}\\",\\"variables\\":{}}",
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
      "Content-Length": 95,
      "Content-Type": "application/json",
      "User-Agent": "axios/0.18.0",
    },
    "maxContentLength": -1,
    "method": "post",
    "timeout": 0,
    "transformRequest": Object {
      "0": [Function],
    },
    "transformResponse": Object {
      "0": [Function],
    },
    "url": "http://localhost:5000",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "data": Object {
      "scalarString": "Hello World",
    },
  },
  "headers": Object {
    "access-control-allow-origin": "*",
    "connection": "close",
    "content-length": "39",
    "content-type": "application/json",
    "date": "Wed, 03 Apr 2019 19:42:44 GMT",
    "x-powered-by": "Express",
  },
  "request": ClientRequest {
    "_contentLength": null,
    "_ended": true,
    "_events": Object {
      "abort": [Function],
      "aborted": [Function],
      "error": [Function],
      "prefinish": [Function],
      "socket": [Function],
      "timeout": [Function],
    },
    "_eventsCount": 6,
    "_hasBody": true,
    "_header": "POST / HTTP/1.1
Accept: application/json, text/plain, */*
Content-Type: application/json
User-Agent: axios/0.18.0
Content-Length: 95
Host: localhost:5000
Connection: close

",
    "_headerSent": true,
    "_last": true,
    "_maxListeners": undefined,
    "_onPendingData": [Function],
    "_redirectable": Writable {
      "_currentRequest": [Circular],
      "_currentUrl": "http://localhost:5000/",
      "_ended": true,
      "_ending": true,
      "_events": Object {
        "error": [Function],
        "response": [Function],
      },
      "_eventsCount": 2,
      "_maxListeners": undefined,
      "_onNativeResponse": [Function],
      "_options": Object {
        "agent": undefined,
        "auth": undefined,
        "headers": Object {
          "Accept": "application/json, text/plain, */*",
          "Content-Length": 95,
          "Content-Type": "application/json",
          "User-Agent": "axios/0.18.0",
        },
        "hostname": "localhost",
        "maxBodyLength": 10485760,
        "maxRedirects": 21,
        "method": "post",
        "nativeProtocols": Object {
          "http:": Object {
            "Agent": [Function],
            "ClientRequest": [Function],
            "IncomingMessage": [Function],
            "METHODS": Array [
              "ACL",
              "BIND",
              "CHECKOUT",
              "CONNECT",
              "COPY",
              "DELETE",
              "GET",
              "HEAD",
              "LINK",
              "LOCK",
              "M-SEARCH",
              "MERGE",
              "MKACTIVITY",
              "MKCALENDAR",
              "MKCOL",
              "MOVE",
              "NOTIFY",
              "OPTIONS",
              "PATCH",
              "POST",
              "PROPFIND",
              "PROPPATCH",
              "PURGE",
              "PUT",
              "REBIND",
              "REPORT",
              "SEARCH",
              "SOURCE",
              "SUBSCRIBE",
              "TRACE",
              "UNBIND",
              "UNLINK",
              "UNLOCK",
              "UNSUBSCRIBE",
            ],
            "OutgoingMessage": [Function],
            "STATUS_CODES": Object {
              "100": "Continue",
              "101": "Switching Protocols",
              "102": "Processing",
              "103": "Early Hints",
              "200": "OK",
              "201": "Created",
              "202": "Accepted",
              "203": "Non-Authoritative Information",
              "204": "No Content",
              "205": "Reset Content",
              "206": "Partial Content",
              "207": "Multi-Status",
              "208": "Already Reported",
              "226": "IM Used",
              "300": "Multiple Choices",
              "301": "Moved Permanently",
              "302": "Found",
              "303": "See Other",
              "304": "Not Modified",
              "305": "Use Proxy",
              "307": "Temporary Redirect",
              "308": "Permanent Redirect",
              "400": "Bad Request",
              "401": "Unauthorized",
              "402": "Payment Required",
              "403": "Forbidden",
              "404": "Not Found",
              "405": "Method Not Allowed",
              "406": "Not Acceptable",
              "407": "Proxy Authentication Required",
              "408": "Request Timeout",
              "409": "Conflict",
              "410": "Gone",
              "411": "Length Required",
              "412": "Precondition Failed",
              "413": "Payload Too Large",
              "414": "URI Too Long",
              "415": "Unsupported Media Type",
              "416": "Range Not Satisfiable",
              "417": "Expectation Failed",
              "418": "I'm a Teapot",
              "421": "Misdirected Request",
              "422": "Unprocessable Entity",
              "423": "Locked",
              "424": "Failed Dependency",
              "425": "Unordered Collection",
              "426": "Upgrade Required",
              "428": "Precondition Required",
              "429": "Too Many Requests",
              "431": "Request Header Fields Too Large",
              "451": "Unavailable For Legal Reasons",
              "500": "Internal Server Error",
              "501": "Not Implemented",
              "502": "Bad Gateway",
              "503": "Service Unavailable",
              "504": "Gateway Timeout",
              "505": "HTTP Version Not Supported",
              "506": "Variant Also Negotiates",
              "507": "Insufficient Storage",
              "508": "Loop Detected",
              "509": "Bandwidth Limit Exceeded",
              "510": "Not Extended",
              "511": "Network Authentication Required",
            },
            "Server": [Function],
            "ServerResponse": [Function],
            "_connectionListener": [Function],
            "createServer": [Function],
            "get": [Function],
            "globalAgent": Agent {
              "_events": Object {
                "free": [Function],
              },
              "_eventsCount": 1,
              "_maxListeners": undefined,
              "defaultPort": 80,
              "freeSockets": Object {},
              "keepAlive": false,
              "keepAliveMsecs": 1000,
              "maxFreeSockets": 256,
              "maxSockets": Infinity,
              "options": Object {
                "path": null,
              },
              "protocol": "http:",
              "requests": Object {},
              "sockets": Object {
                "localhost:5000:": Array [
                  Socket {
                    "_events": Object {
                      "agentRemove": [Function],
                      "close": Array [
                        [Function],
                        [Function],
                      ],
                      "drain": [Function],
                      "end": [Function],
                      "error": [Function],
                      "finish": [Function],
                      "free": [Function],
                    },
                    "_eventsCount": 7,
                    "_hadError": false,
                    "_handle": TCP {
                      "onconnection": null,
                      "onread": [Function],
                      "reading": true,
                      Symbol(owner): [Circular],
                    },
                    "_host": "localhost",
                    "_httpMessage": [Circular],
                    "_maxListeners": undefined,
                    "_parent": null,
                    "_pendingData": null,
                    "_pendingEncoding": "",
                    "_readableState": ReadableState {
                      "awaitDrain": 0,
                      "buffer": BufferList {
                        "head": null,
                        "length": 0,
                        "tail": null,
                      },
                      "decoder": null,
                      "defaultEncoding": "utf8",
                      "destroyed": false,
                      "emitClose": false,
                      "emittedReadable": false,
                      "encoding": null,
                      "endEmitted": false,
                      "ended": false,
                      "flowing": true,
                      "highWaterMark": 16384,
                      "length": 0,
                      "needReadable": true,
                      "objectMode": false,
                      "paused": false,
                      "pipes": null,
                      "pipesCount": 0,
                      "readableListening": false,
                      "reading": true,
                      "readingMore": false,
                      "resumeScheduled": false,
                      "sync": false,
                    },
                    "_server": null,
                    "_sockname": null,
                    "_writableState": WritableState {
                      "bufferProcessing": false,
                      "bufferedRequest": null,
                      "bufferedRequestCount": 0,
                      "corked": 0,
                      "corkedRequestsFree": Object {
                        "entry": null,
                        "finish": [Function],
                        "next": Object {
                          "entry": null,
                          "finish": [Function],
                          "next": null,
                        },
                      },
                      "decodeStrings": false,
                      "defaultEncoding": "utf8",
                      "destroyed": false,
                      "emitClose": false,
                      "ended": true,
                      "ending": true,
                      "errorEmitted": false,
                      "finalCalled": true,
                      "finished": false,
                      "highWaterMark": 16384,
                      "lastBufferedRequest": null,
                      "length": 0,
                      "needDrain": false,
                      "objectMode": false,
                      "onwrite": [Function],
                      "pendingcb": 1,
                      "prefinished": false,
                      "sync": false,
                      "writecb": null,
                      "writelen": 0,
                      "writing": false,
                    },
                    "allowHalfOpen": false,
                    "connecting": false,
                    "parser": null,
                    "readable": true,
                    "server": null,
                    "writable": false,
                    Symbol(asyncId): 11,
                    Symbol(lastWriteQueueSize): 0,
                    Symbol(timeout): null,
                    Symbol(kBytesRead): 0,
                    Symbol(kBytesWritten): 0,
                  },
                ],
              },
            },
            "maxHeaderSize": 8192,
            "request": [Function],
          },
          "https:": Object {
            "Agent": [Function],
            "Server": [Function],
            "createServer": [Function],
            "get": [Function],
            "globalAgent": Agent {
              "_events": Object {
                "free": [Function],
              },
              "_eventsCount": 1,
              "_maxListeners": undefined,
              "_sessionCache": Object {
                "list": Array [],
                "map": Object {},
              },
              "defaultPort": 443,
              "freeSockets": Object {},
              "keepAlive": false,
              "keepAliveMsecs": 1000,
              "maxCachedSessions": 100,
              "maxFreeSockets": 256,
              "maxSockets": Infinity,
              "options": Object {
                "path": null,
              },
              "protocol": "https:",
              "requests": Object {},
              "sockets": Object {},
            },
            "request": [Function],
          },
        },
        "path": "/",
        "pathname": "/",
        "port": "5000",
        "protocol": "http:",
      },
      "_redirectCount": 0,
      "_redirects": Array [],
      "_requestBodyBuffers": Array [],
      "_requestBodyLength": 95,
      "_writableState": WritableState {
        "bufferProcessing": false,
        "bufferedRequest": null,
        "bufferedRequestCount": 0,
        "corked": 0,
        "corkedRequestsFree": Object {
          "entry": null,
          "finish": [Function],
          "next": null,
        },
        "decodeStrings": true,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": true,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "lastBufferedRequest": null,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "writable": true,
    },
    "_removedConnection": false,
    "_removedContLen": false,
    "_removedTE": false,
    "_trailer": "",
    "aborted": undefined,
    "agent": Agent {
      "_events": Object {
        "free": [Function],
      },
      "_eventsCount": 1,
      "_maxListeners": undefined,
      "defaultPort": 80,
      "freeSockets": Object {},
      "keepAlive": false,
      "keepAliveMsecs": 1000,
      "maxFreeSockets": 256,
      "maxSockets": Infinity,
      "options": Object {
        "path": null,
      },
      "protocol": "http:",
      "requests": Object {},
      "sockets": Object {
        "localhost:5000:": Array [
          Socket {
            "_events": Object {
              "agentRemove": [Function],
              "close": Array [
                [Function],
                [Function],
              ],
              "drain": [Function],
              "end": [Function],
              "error": [Function],
              "finish": [Function],
              "free": [Function],
            },
            "_eventsCount": 7,
            "_hadError": false,
            "_handle": TCP {
              "onconnection": null,
              "onread": [Function],
              "reading": true,
              Symbol(owner): [Circular],
            },
            "_host": "localhost",
            "_httpMessage": [Circular],
            "_maxListeners": undefined,
            "_parent": null,
            "_pendingData": null,
            "_pendingEncoding": "",
            "_readableState": ReadableState {
              "awaitDrain": 0,
              "buffer": BufferList {
                "head": null,
                "length": 0,
                "tail": null,
              },
              "decoder": null,
              "defaultEncoding": "utf8",
              "destroyed": false,
              "emitClose": false,
              "emittedReadable": false,
              "encoding": null,
              "endEmitted": false,
              "ended": false,
              "flowing": true,
              "highWaterMark": 16384,
              "length": 0,
              "needReadable": true,
              "objectMode": false,
              "paused": false,
              "pipes": null,
              "pipesCount": 0,
              "readableListening": false,
              "reading": true,
              "readingMore": false,
              "resumeScheduled": false,
              "sync": false,
            },
            "_server": null,
            "_sockname": null,
            "_writableState": WritableState {
              "bufferProcessing": false,
              "bufferedRequest": null,
              "bufferedRequestCount": 0,
              "corked": 0,
              "corkedRequestsFree": Object {
                "entry": null,
                "finish": [Function],
                "next": Object {
                  "entry": null,
                  "finish": [Function],
                  "next": null,
                },
              },
              "decodeStrings": false,
              "defaultEncoding": "utf8",
              "destroyed": false,
              "emitClose": false,
              "ended": true,
              "ending": true,
              "errorEmitted": false,
              "finalCalled": true,
              "finished": false,
              "highWaterMark": 16384,
              "lastBufferedRequest": null,
              "length": 0,
              "needDrain": false,
              "objectMode": false,
              "onwrite": [Function],
              "pendingcb": 1,
              "prefinished": false,
              "sync": false,
              "writecb": null,
              "writelen": 0,
              "writing": false,
            },
            "allowHalfOpen": false,
            "connecting": false,
            "parser": null,
            "readable": true,
            "server": null,
            "writable": false,
            Symbol(asyncId): 11,
            Symbol(lastWriteQueueSize): 0,
            Symbol(timeout): null,
            Symbol(kBytesRead): 0,
            Symbol(kBytesWritten): 0,
          },
        ],
      },
    },
    "chunkedEncoding": false,
    "connection": Socket {
      "_events": Object {
        "agentRemove": [Function],
        "close": Array [
          [Function],
          [Function],
        ],
        "drain": [Function],
        "end": [Function],
        "error": [Function],
        "finish": [Function],
        "free": [Function],
      },
      "_eventsCount": 7,
      "_hadError": false,
      "_handle": TCP {
        "onconnection": null,
        "onread": [Function],
        "reading": true,
        Symbol(owner): [Circular],
      },
      "_host": "localhost",
      "_httpMessage": [Circular],
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "awaitDrain": 0,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "flowing": true,
        "highWaterMark": 16384,
        "length": 0,
        "needReadable": true,
        "objectMode": false,
        "paused": false,
        "pipes": null,
        "pipesCount": 0,
        "readableListening": false,
        "reading": true,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": false,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "bufferProcessing": false,
        "bufferedRequest": null,
        "bufferedRequestCount": 0,
        "corked": 0,
        "corkedRequestsFree": Object {
          "entry": null,
          "finish": [Function],
          "next": Object {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
        },
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": true,
        "ending": true,
        "errorEmitted": false,
        "finalCalled": true,
        "finished": false,
        "highWaterMark": 16384,
        "lastBufferedRequest": null,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 1,
        "prefinished": false,
        "sync": false,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "allowHalfOpen": false,
      "connecting": false,
      "parser": null,
      "readable": true,
      "server": null,
      "writable": false,
      Symbol(asyncId): 11,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "finished": true,
    "maxHeadersCount": null,
    "method": "POST",
    "output": Array [],
    "outputCallbacks": Array [],
    "outputEncodings": Array [],
    "outputSize": 0,
    "parser": null,
    "path": "/",
    "res": IncomingMessage {
      "_consuming": false,
      "_dumped": false,
      "_events": Object {
        "data": [Function],
        "end": Array [
          [Function],
          [Function],
        ],
        "error": [Function],
      },
      "_eventsCount": 3,
      "_maxListeners": undefined,
      "_readableState": ReadableState {
        "awaitDrain": 0,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": true,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": true,
        "ended": true,
        "flowing": true,
        "highWaterMark": 16384,
        "length": 0,
        "needReadable": false,
        "objectMode": false,
        "paused": false,
        "pipes": null,
        "pipesCount": 0,
        "readableListening": false,
        "reading": false,
        "readingMore": true,
        "resumeScheduled": false,
        "sync": true,
      },
      "aborted": false,
      "client": Socket {
        "_events": Object {
          "agentRemove": [Function],
          "close": Array [
            [Function],
            [Function],
          ],
          "drain": [Function],
          "end": [Function],
          "error": [Function],
          "finish": [Function],
          "free": [Function],
        },
        "_eventsCount": 7,
        "_hadError": false,
        "_handle": TCP {
          "onconnection": null,
          "onread": [Function],
          "reading": true,
          Symbol(owner): [Circular],
        },
        "_host": "localhost",
        "_httpMessage": [Circular],
        "_maxListeners": undefined,
        "_parent": null,
        "_pendingData": null,
        "_pendingEncoding": "",
        "_readableState": ReadableState {
          "awaitDrain": 0,
          "buffer": BufferList {
            "head": null,
            "length": 0,
            "tail": null,
          },
          "decoder": null,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "emittedReadable": false,
          "encoding": null,
          "endEmitted": false,
          "ended": false,
          "flowing": true,
          "highWaterMark": 16384,
          "length": 0,
          "needReadable": true,
          "objectMode": false,
          "paused": false,
          "pipes": null,
          "pipesCount": 0,
          "readableListening": false,
          "reading": true,
          "readingMore": false,
          "resumeScheduled": false,
          "sync": false,
        },
        "_server": null,
        "_sockname": null,
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": Object {
            "entry": null,
            "finish": [Function],
            "next": Object {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
          },
          "decodeStrings": false,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "ended": true,
          "ending": true,
          "errorEmitted": false,
          "finalCalled": true,
          "finished": false,
          "highWaterMark": 16384,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": false,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "allowHalfOpen": false,
        "connecting": false,
        "parser": null,
        "readable": true,
        "server": null,
        "writable": false,
        Symbol(asyncId): 11,
        Symbol(lastWriteQueueSize): 0,
        Symbol(timeout): null,
        Symbol(kBytesRead): 0,
        Symbol(kBytesWritten): 0,
      },
      "complete": true,
      "connection": Socket {
        "_events": Object {
          "agentRemove": [Function],
          "close": Array [
            [Function],
            [Function],
          ],
          "drain": [Function],
          "end": [Function],
          "error": [Function],
          "finish": [Function],
          "free": [Function],
        },
        "_eventsCount": 7,
        "_hadError": false,
        "_handle": TCP {
          "onconnection": null,
          "onread": [Function],
          "reading": true,
          Symbol(owner): [Circular],
        },
        "_host": "localhost",
        "_httpMessage": [Circular],
        "_maxListeners": undefined,
        "_parent": null,
        "_pendingData": null,
        "_pendingEncoding": "",
        "_readableState": ReadableState {
          "awaitDrain": 0,
          "buffer": BufferList {
            "head": null,
            "length": 0,
            "tail": null,
          },
          "decoder": null,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "emittedReadable": false,
          "encoding": null,
          "endEmitted": false,
          "ended": false,
          "flowing": true,
          "highWaterMark": 16384,
          "length": 0,
          "needReadable": true,
          "objectMode": false,
          "paused": false,
          "pipes": null,
          "pipesCount": 0,
          "readableListening": false,
          "reading": true,
          "readingMore": false,
          "resumeScheduled": false,
          "sync": false,
        },
        "_server": null,
        "_sockname": null,
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": Object {
            "entry": null,
            "finish": [Function],
            "next": Object {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
          },
          "decodeStrings": false,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "ended": true,
          "ending": true,
          "errorEmitted": false,
          "finalCalled": true,
          "finished": false,
          "highWaterMark": 16384,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": false,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "allowHalfOpen": false,
        "connecting": false,
        "parser": null,
        "readable": true,
        "server": null,
        "writable": false,
        Symbol(asyncId): 11,
        Symbol(lastWriteQueueSize): 0,
        Symbol(timeout): null,
        Symbol(kBytesRead): 0,
        Symbol(kBytesWritten): 0,
      },
      "headers": Object {
        "access-control-allow-origin": "*",
        "connection": "close",
        "content-length": "39",
        "content-type": "application/json",
        "date": "Wed, 03 Apr 2019 19:42:44 GMT",
        "x-powered-by": "Express",
      },
      "httpVersion": "1.1",
      "httpVersionMajor": 1,
      "httpVersionMinor": 1,
      "method": null,
      "rawHeaders": Array [
        "X-Powered-By",
        "Express",
        "Access-Control-Allow-Origin",
        "*",
        "Content-Type",
        "application/json",
        "Content-Length",
        "39",
        "Date",
        "Wed, 03 Apr 2019 19:42:44 GMT",
        "Connection",
        "close",
      ],
      "rawTrailers": Array [],
      "readable": false,
      "redirects": Array [],
      "req": [Circular],
      "responseUrl": "http://localhost:5000/",
      "socket": Socket {
        "_events": Object {
          "agentRemove": [Function],
          "close": Array [
            [Function],
            [Function],
          ],
          "drain": [Function],
          "end": [Function],
          "error": [Function],
          "finish": [Function],
          "free": [Function],
        },
        "_eventsCount": 7,
        "_hadError": false,
        "_handle": TCP {
          "onconnection": null,
          "onread": [Function],
          "reading": true,
          Symbol(owner): [Circular],
        },
        "_host": "localhost",
        "_httpMessage": [Circular],
        "_maxListeners": undefined,
        "_parent": null,
        "_pendingData": null,
        "_pendingEncoding": "",
        "_readableState": ReadableState {
          "awaitDrain": 0,
          "buffer": BufferList {
            "head": null,
            "length": 0,
            "tail": null,
          },
          "decoder": null,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "emittedReadable": false,
          "encoding": null,
          "endEmitted": false,
          "ended": false,
          "flowing": true,
          "highWaterMark": 16384,
          "length": 0,
          "needReadable": true,
          "objectMode": false,
          "paused": false,
          "pipes": null,
          "pipesCount": 0,
          "readableListening": false,
          "reading": true,
          "readingMore": false,
          "resumeScheduled": false,
          "sync": false,
        },
        "_server": null,
        "_sockname": null,
        "_writableState": WritableState {
          "bufferProcessing": false,
          "bufferedRequest": null,
          "bufferedRequestCount": 0,
          "corked": 0,
          "corkedRequestsFree": Object {
            "entry": null,
            "finish": [Function],
            "next": Object {
              "entry": null,
              "finish": [Function],
              "next": null,
            },
          },
          "decodeStrings": false,
          "defaultEncoding": "utf8",
          "destroyed": false,
          "emitClose": false,
          "ended": true,
          "ending": true,
          "errorEmitted": false,
          "finalCalled": true,
          "finished": false,
          "highWaterMark": 16384,
          "lastBufferedRequest": null,
          "length": 0,
          "needDrain": false,
          "objectMode": false,
          "onwrite": [Function],
          "pendingcb": 1,
          "prefinished": false,
          "sync": false,
          "writecb": null,
          "writelen": 0,
          "writing": false,
        },
        "allowHalfOpen": false,
        "connecting": false,
        "parser": null,
        "readable": true,
        "server": null,
        "writable": false,
        Symbol(asyncId): 11,
        Symbol(lastWriteQueueSize): 0,
        Symbol(timeout): null,
        Symbol(kBytesRead): 0,
        Symbol(kBytesWritten): 0,
      },
      "statusCode": 200,
      "statusMessage": "OK",
      "trailers": Object {},
      "upgrade": false,
      "url": "",
    },
    "sendDate": false,
    "shouldKeepAlive": false,
    "socket": Socket {
      "_events": Object {
        "agentRemove": [Function],
        "close": Array [
          [Function],
          [Function],
        ],
        "drain": [Function],
        "end": [Function],
        "error": [Function],
        "finish": [Function],
        "free": [Function],
      },
      "_eventsCount": 7,
      "_hadError": false,
      "_handle": TCP {
        "onconnection": null,
        "onread": [Function],
        "reading": true,
        Symbol(owner): [Circular],
      },
      "_host": "localhost",
      "_httpMessage": [Circular],
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "awaitDrain": 0,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "flowing": true,
        "highWaterMark": 16384,
        "length": 0,
        "needReadable": true,
        "objectMode": false,
        "paused": false,
        "pipes": null,
        "pipesCount": 0,
        "readableListening": false,
        "reading": true,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": false,
      },
      "_server": null,
      "_sockname": null,
      "_writableState": WritableState {
        "bufferProcessing": false,
        "bufferedRequest": null,
        "bufferedRequestCount": 0,
        "corked": 0,
        "corkedRequestsFree": Object {
          "entry": null,
          "finish": [Function],
          "next": Object {
            "entry": null,
            "finish": [Function],
            "next": null,
          },
        },
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": true,
        "ending": true,
        "errorEmitted": false,
        "finalCalled": true,
        "finished": false,
        "highWaterMark": 16384,
        "lastBufferedRequest": null,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 1,
        "prefinished": false,
        "sync": false,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "allowHalfOpen": false,
      "connecting": false,
      "parser": null,
      "readable": true,
      "server": null,
      "writable": false,
      Symbol(asyncId): 11,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "socketPath": undefined,
    "timeout": undefined,
    "timeoutCb": null,
    "upgradeOrConnect": false,
    "useChunkedEncodingByDefault": true,
    "writable": true,
    Symbol(isCorked): false,
    Symbol(outHeadersKey): Object {
      "accept": Array [
        "Accept",
        "application/json, text/plain, */*",
      ],
      "content-length": Array [
        "Content-Length",
        95,
      ],
      "content-type": Array [
        "Content-Type",
        "application/json",
      ],
      "host": Array [
        "Host",
        "localhost:5000",
      ],
      "user-agent": Array [
        "User-Agent",
        "axios/0.18.0",
      ],
    },
  },
  "status": 200,
  "statusText": "OK",
}
`;
